lib/src/signed_video_tlv.c

```diff
@@ -236,9 +236,8 @@ encode_general(signed_video_t *self, uint8_t *data)
   if (gop_info->has_timestamp) {
     data_size += sizeof(timestamp);
   }
-  if (self->linked_hash_on) {
-    data_size += gop_info->linked_hash_data.hash_size;
-  }
+  data_size += gop_info->linked_hash_data.hash_size;
```

Comment:
The hash size is available in `sign_data`. No need to hold a separate member for that.